In [10]:
import pandas as pd


In [11]:
df = pd.read_csv("sales_data.csv")
print(df)

   order_id region   product  quantity  price
0      1001   East  Keyboard       2.0   1500
1      1002   West     Mouse       5.0    500
2      1003   East   Monitor       NaN  12000
3      1004  South  Keyboard       1.0   1500
4      1005   West   Monitor       2.0  12000


Which columns contain missing values?
# 1. Missing values: quantity column has 1 missing value (row 1003)
What are the numeric columns?
# 2. Numeric columns: order_id, quantity, price
What potential data issues do you see?
# 3. Potential issues: 
#    - Missing quantity for Monitor order
#    - Need to verify if price values are reasonable

In [12]:
df.head()
df.info()
df.describe()

<class 'pandas.DataFrame'>
RangeIndex: 5 entries, 0 to 4
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   order_id  5 non-null      int64  
 1   region    5 non-null      str    
 2   product   5 non-null      str    
 3   quantity  4 non-null      float64
 4   price     5 non-null      int64  
dtypes: float64(1), int64(2), str(2)
memory usage: 332.0 bytes


order_id  quantity         price
count     5.000000  4.000000      5.000000
mean   1003.000000  2.500000   5500.000000
std       1.581139  1.732051   5947.688627
min    1001.000000  1.000000    500.000000
25%    1002.000000  1.750000   1500.000000
50%    1003.000000  2.000000   1500.000000
75%    1004.000000  2.750000  12000.000000
max    1005.000000  5.000000  12000.000000

In [13]:
#Extract a column as a Series
region_series = df["region"]
print(type(region_series))
#compare with dataframe type
print(type(df))
# inspect the series
print(region_series)



<class 'pandas.Series'>
<class 'pandas.DataFrame'>
0     East
1     West
2     East
3    South
4     West
Name: region, dtype: str


In [17]:
#check missing values in data frame
df.isnull().sum()




order_id    0
region      0
product     0
quantity    1
price       0
dtype: int64

In [18]:
df.shape

(5, 5)

In [16]:
#get column names
df.columns

Index(['order_id', 'region', 'product', 'quantity', 'price'], dtype='str')

In [19]:
df.dtypes


order_id      int64
region          str
product         str
quantity    float64
price         int64
dtype: object